projects/plugins/boost/app/features/optimizations/cloud-css/Cloud_CSS_Request.php

```diff
@@ -28,16 +28,29 @@ public function __construct( $state ) {
 	}
 
 	public function request_generate() {
-		$sources = $this->state->get_provider_urls();
+		$sources   = $this->state->get_provider_urls();
+		$providers = $this->get_css_generate_providers( $sources );
 
 		$client               = Boost_API::get_client();
-		$payload              = array( 'providers' => $sources );
+		$payload              = array( 'providers' => $providers );
 		$payload['requestId'] = md5( wp_json_encode( $payload ) );
 
 		$this->reset_existing_css();
 		return $client->post( 'cloud-css', $payload );
 	}
 
+	private function get_css_generate_providers( $providers ) {
+		$formatted = array();
+		foreach ( $providers as $key => $urls ) {
+			$generate_urls = array();
+			foreach ( $urls as $url ) {
+				$generate_urls[] = add_query_arg( 'jb-generate-critical-css', 'true', $url );
```

Comment:
I have just noticed something. Because the local and cloud critical CSS handle adding the parameter in a different way, cloud CSS doesn't remove the parameter while displaying the error. I think we should remove it before displaying. But, it is of lower priority and we can do it in a different PR.

Btw, what's the advantage of having this param for CCSS?

![](https://d.pr/i/S1mQqC+)

Reply:
You are right. This is only used for testing purposes. I do not see the necessity to keep this or at least we need to remove the arg. However, didn't want to keep this open for long, hence merged. I will update it with the advanced recommendation PR.

Reply:
This is now included in #23919